python/tk_multi_workfiles/browser_form.py

```diff
@@ -669,3 +675,40 @@ def _on_step_filter_changed(self, step_list):
         :param step_list: A list of Shotgun Step dictionaries.
         """
         self.step_filter_changed.emit(get_filter_from_filter_list(step_list))
+
+    # Task status filter methods
+    def populate_task_status_list(self):
+        """
+        Populate the task status list from the project schema.
+        """
+        app = sgtk.platform.current_bundle()
+        project_id = app.context.project.get("id")
+        if project_id:
+            logger.debug(
+                f"Task Status Filter: getting statuses from project {project_id}"
+            )
+            try:
+                task_status_list = app.shotgun._sg.schema_field_read(
+                    "Task", "sg_status_list", {"type": "Project", "id": project_id}
+                )["sg_status_list"]["properties"]["valid_values"]["value"]
+            except KeyError:
+                task_status_list = []
```

Comment:
Do we need to populate a single "ALL" entry in this case? That does not make sense. It'd be better to hide/grey the widget in this case